elastic/elastic-agent
# testing/runner: use sudo rm on host machine and handle errors

When tests are executed with sudo, the fetched artefact and other created files are owned by root. Therefore to successfully remove the agent folder it must be done with sudo.



## What does this PR do?

Remove files on the integration tests host machine with sudo and handle the possible errors

## Why is it important?

When the test is run with sudo, some files will belong to root, which caused the removal of the `agent` folder to fail. Also handling the errors will surface any other problem that might still happen.

## Checklist




## How to test this PR locally

run any integration test using the local fetcher and the SNAPSHOT version. Rum it at least once . Rebuild the agent in a was the .sha512 file will change (or just change the file). Run the test again. 

Without this fix you should find in the host machine the `~/agent/.agent-cache/` files were not updated whereas the artefacts were copied by the test runner. On the home folder there will be an agent and it's sha512 file, this one should be the new one, whereas the `~/agent/.agent-cache/` is still the previous one.

## Related issues

N/A

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
